File: frontend/src/types/exercise.ts
export type DifficultyLevel = "easy" | "medium" | "hard"
export type ExerciseType = "read_aloud" | "free_speech" | "qa" | "craa"

export interface Exercise {
  id: number
  teacher_id: number
  title: string
  description: string
  reference_text?: string | null
  difficulty: DifficultyLevel
  exercise_type: ExerciseType
  created_at?: string | null
  teacher_name?: string | null
  argument_text?: string | null
  topic_context?: string | null
  key_claim?: string | null
  preparation_time?: number | null
  response_time?: number | null
  has_argument_audio?: boolean
  video_url?: string | null
}

export interface CreateExerciseInput {
  title: string
  description: string
  reference_text?: string
  difficulty: DifficultyLevel
  exercise_type: ExerciseType
  argument_text?: string
  topic_context?: string
  key_claim?: string
  preparation_time?: number
  response_time?: number
  video_url?: string
}

File: backend/app/models/exercise.py
from pydantic import BaseModel
from typing import Optional
from enum import Enum


class DifficultyLevel(str, Enum):
    easy = "easy"
    medium = "medium"
    hard = "hard"


class ExerciseType(str, Enum):
    read_aloud = "read_aloud"
    free_speech = "free_speech"
    qa = "qa"
    craa = "craa"


class CreateExerciseInput(BaseModel):
    title: str
    description: str
    reference_text: Optional[str] = None
    difficulty: DifficultyLevel = DifficultyLevel.medium
    exercise_type: ExerciseType = ExerciseType.free_speech
    argument_text: Optional[str] = None
    topic_context: Optional[str] = None
    key_claim: Optional[str] = None
    preparation_time: Optional[int] = 120
    response_time: Optional[int] = 120
    video_url: Optional[str] = None


class UpdateExerciseInput(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    reference_text: Optional[str] = None
    difficulty: Optional[DifficultyLevel] = None
    exercise_type: Optional[ExerciseType] = None
    argument_text: Optional[str] = None
    topic_context: Optional[str] = None
    key_claim: Optional[str] = None
    preparation_time: Optional[int] = None
    response_time: Optional[int] = None
    video_url: Optional[str] = None


class ExerciseResponse(BaseModel):
    id: int
    teacher_id: int
    title: str
    description: str
    reference_text: Optional[str] = None
    difficulty: str
    exercise_type: str
    created_at: Optional[str] = None
    teacher_name: Optional[str] = None
    argument_text: Optional[str] = None
    topic_context: Optional[str] = None
    key_claim: Optional[str] = None
    preparation_time: Optional[int] = None
    response_time: Optional[int] = None
    has_argument_audio: Optional[bool] = False
    video_url: Optional[str] = None
